pallets/feature-control/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+//! # Feature control pallet
+//!
+//! This pallet provides a way of turning on/off features in the runtime that cannot be controlled with runtime
+//! configuration. It maintains a simple map of feature identifiers together with their status (enabled/disabled). It is
+//! supposed to be modified only by the specified origin, but read by any runtime code.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit = "256"]
+#![deny(missing_docs)]
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+#[cfg(test)]
+mod tests;
+mod weights;
+
+use frame_support::pallet_prelude::StorageVersion;
+pub use pallet::*;
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_core::RuntimeDebug;
+pub use weights::{AlephWeight, WeightInfo};
+
+/// All available optional features for the Aleph Zero runtime.
+#[derive(Clone, Copy, PartialEq, Eq, RuntimeDebug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+pub enum Feature {
+    /// The on-chain verifier feature involves:
+    /// - VkStorage pallet (for storing verification keys)
+    /// - smart contract chain extension exposing `verify` function
+    /// - SnarkVerifier runtime interface
+    #[codec(index = 0)]
+    OnChainVerifier,
+}
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::OriginFor;
+
+    use super::{weights::WeightInfo, *};
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Item required for emitting events.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+        /// Weight information for the pallet's extrinsics.
+        type WeightInfo: WeightInfo;
+        /// The origin that can modify the feature map.
+        type Controller: EnsureOrigin<Self::RuntimeOrigin>;
```

Comment:
nit: Controller may be a little confusing given we had controllers in the staking pallet.